revapi-java/src/main/java-mr/9/org/revapi/java/compilation/Compiler.java

```diff
@@ -0,0 +1,241 @@
+/*
```

Comment:
Instead of duplicating the whole class and changing just the way the compiler options are composed, I think it'd be better to introduce a sort of package-private base class that would implement all of the common logic (i.e. basically a copy of what Compiler class currently is) and just define a new package-private abstract method for getting the compiler options that each of the Compiler classes would implement according to the java compiler version requirements (i.e. the Compiler in the main sources would use the classpath the old way and the Compiler in 9 would use the module path as you changed it).

Reply:
Yes, I agree. We will refactor this in order to implement it in that way.